train.py

```diff
@@ -46,21 +47,50 @@ def loss_fn(y_true, y_pred):
     return tf.reduce_mean(nll)
 
 
-def crop_3d():
-    crop_size = 20
+class ReparamTrickLayer(keras.layers.Layer):
+    def call(self, input, *args, **kwargs):
+        oef_sample = input[:, :, :, :, 0] + tf.random.normal(tf.shape(input[:, :, :, :, 0])) * tf.exp(
+            input[:, :, :, :, 1])
+        dbv_sample = input[:, :, :, :, 2] + tf.random.normal(tf.shape(input[:, :, :, :, 0])) * tf.exp(
```

Comment:
Should this be
 `dbv_sample = input[:, :, :, :, 2] + tf.random.normal(tf.shape(input[:, :, :, :, 2]))` * ...

Reply:
Found it in [the paper](https://arxiv.org/pdf/1312.6114.pdf): 2.4 The reparameterization trick

Reply:
I'm just taking the shape of it and it's the same for all of them. It'd be cleaner if we just saved the shape for the samples and passed it into both of them